go/kbfs/libkbfs/prefetcher.go

```diff
@@ -390,10 +396,16 @@ func (p *blockPrefetcher) request(ctx context.Context, priority int,
 	if !isPrefetchWaiting {
 		// If the block isn't in the tree, we add it with a block count of 1 (a
 		// later TriggerPrefetch will come in and decrement it).
+		obseleted := make(chan struct{})
 		req := &prefetchRequest{ptr, block.NewEmptier(), kmd, priority,
-			lifetime, NoPrefetch, action, nil}
+			lifetime, NoPrefetch, action, nil, obseleted}
 		pre = p.newPrefetch(1, false, req)
 		p.prefetches[ptr.ID] = pre
+		p.queuedPrefetchHandlesLock.Lock()
+		// TODO: figure out what to do in the event that there is already a
```

Comment:
This is a pretty big `TODO`; we shouldn't merge without this being figured out.

Reply:
Yeah I think we should either keep a slice of channels in `queuedPrefetchHandles`, or keep the full channel in there so we can reuse it in the `prefetchRequest` above if it already exists.

Reply:
I prefer the latter solution, but each of them seems to have a problem w/ deletion:
 - if multiple queued prefetches are using the same channel then we can't know when to delete the channel from `queuedPrefetchHandles` and we get a memory leak
 - if we use a slice, then deleting them gets complicated.

Reply:
Hmm.  I guess you can keep a struct that contains both the single channel and a counter.  Increment the counter on each `triggerPrefetch`, and decrement it when the request is processed.

Reply:
(And obviously when the counter is 0, you can clean up the map entry.)